tests/lib/test_docker.py

```diff
@@ -58,3 +59,40 @@ def run(args, timeout, **kwargs):
     monkeypatch.setattr(docker, "subprocess_run", error(None, msg))
     with pytest.raises(docker.DockerDiskSpaceError):
         docker.docker([])
+
+
+@pytest.mark.needs_docker
+def test_copy_from_volume(tmp_path, docker_cleanup):
+    volume = __name__
+    src = tmp_path / "src.txt"
+    dst = tmp_path / "dst.txt"
+    tmp = dst.with_suffix(dst.suffix + ".tmp")
+    src.write_text("I exist")
+    docker.create_volume(volume)
+    docker.copy_to_volume(volume, src, "src.txt")
+    docker.copy_from_volume(volume, "src.txt", dst)
+    assert dst.read_text() == "I exist"
+    assert not tmp.exists()
```

Comment:
You'll need to do a `glob` here to check there's nothing matching the pattern `*.tmp` as the temporary filename is no longer completely predictable.

Reply:
Fixed